mucommander-core/src/main/java/com/mucommander/ui/theme/ThemeReader.java

```diff
@@ -596,203 +666,256 @@ else if(state == STATE_QUICK_LIST_ITEM_SELECTED)
+                        state = STATE_TABLE;
+                        break;
+                    case STATE_QUICK_LIST_ITEM_NORMAL:
+                        state = STATE_QUICK_LIST_ITEM;
+                        break;
+                }
+                break;
+            // Selected element declaration.
+            case ELEMENT_SELECTED:
+                switch (state) {
+                    case STATE_TERMINAL_SELECTED:
+                        state = STATE_TERMINAL;
+                        break;
+                    case STATE_HIDDEN_SELECTED:
+                        state = STATE_HIDDEN;
+                        break;
+                    case STATE_FOLDER_SELECTED:
+                        state = STATE_FOLDER;
+                        break;
+                    case STATE_ARCHIVE_SELECTED:
+                        state = STATE_ARCHIVE;
+                        break;
+                    case STATE_SYMLINK_SELECTED:
+                        state = STATE_SYMLINK;
+                        break;
+                    case STATE_READ_ONLY_SELECTED:
+                        state = STATE_READ_ONLY;
+                        break;
+                    case STATE_MARKED_SELECTED:
+                        state = STATE_MARKED;
+                        break;
+                    case STATE_FILE_SELECTED:
+                        state = STATE_FILE;
+                        break;
+                    case STATE_EDITOR_SELECTED:
+                        state = STATE_EDITOR;
+                        break;
+                    case STATE_LOCATION_BAR_SELECTED:
+                        state = STATE_LOCATION_BAR;
+                        break;
+                    case STATE_TABLE_SELECTED:
+                        state = STATE_TABLE;
+                        break;
+                    case STATE_QUICK_LIST_ITEM_SELECTED:
+                        state = STATE_QUICK_LIST_ITEM;
+                        break;
+                }
+                break;
         }
     }
 
+    public static void preSetAvailableFonts(Set<String> fontCache) {
+        if (!fontCache.isEmpty()) {
+            availableFonts = fontCache;
+        }
+        new Timer("RefreshFontNames").schedule(new TimerTask() {
+            @Override
+            public void run() {
+                LOGGER.info("Going to refresh available fonts from OS....");
+                availableFonts = getAvailableFontsFromOS();
+            }
+        }, 10 * 1000L);
+    }
 
+    public static Set<String> getAvailableFonts() {
+        if (availableFonts == null) {
+            availableFonts = getAvailableFontsFromOS();
+        }
+        return availableFonts;
+    }
 
     // - Helper methods ------------------------------------------------------
     // -----------------------------------------------------------------------
+
+    private static Set<String> getAvailableFontsFromOS() {
+        return new HashSet<>(List.of(
+                        // takes long time....
```

Comment:
yeah, it makes sense to cache but the way it's implemented now, after starting the app we would never get new fonts that are installed after the app started, no?

Reply:
do you mean while app is running we won't see new fonts? Or between restarts?

Reply:
I meant between restarts
I think it's ok though if we'll store the cache in the snapshot file